tests/snuba/test_organization_discover_saved_query_detail.py

```diff
@@ -57,3 +57,71 @@ def test_get_org_without_access(self):
             response = self.client.get(url)
 
         assert response.status_code == 403, response.content
+
+    def test_put(self):
```

Comment:
What about updating a query that doesn't exist or you can't modify?

Reply:
Good point, add some tests for these scenarios